gen/build_deploy/bash.py

```diff
@@ -668,33 +668,28 @@ def fill_template(base_name, format_args):
     return installer_filename
 
 
-def get_launch_package_id(launch_info):
+def get_launch_package_id(package_list):
+    """scans througha the package list of the
+    """
     launch_package_id = None
-    for package_id in launch_info['packages']:
+    for package_id in package_list:
         if pkgpanda.PackageId(package_id).name == 'dcos-launch':
             launch_package_id = package_id
             break
     assert launch_package_id, 'dcos-launch package cannot be found!'
     return launch_package_id
 
 
-def fetch_dcos_launch_bin(launch_name, util_bootstrap_filename, launch_package_id):
-    """ Takes a util bootstrap tarball and unpacks it, finds the specific
-    dcos-launch package included with tarball, and extracts the binary from that.
-    In theory, one could just find the active package, however symlinks are broken
-    when the tarball is unpacked because it is not being unpacked into a DC/OS env.
-    By checking the specific package id, we avoid grabbing something unintentionally
-    """
+def fetch_dcos_launch_bin(launch_name, launch_package_id):
     work_dir = py.path.local.mkdtemp()
-    unpacked_bin_path = str(work_dir.join('packages/{}/dcos-launch'.format(launch_package_id)))
-    cached_util_bootstrap = os.getcwd() + '/packages/cache/bootstrap/' + util_bootstrap_filename
-    temp_util_bootstrap = str(work_dir.join(util_bootstrap_filename))
-    subprocess.check_call(['cp', cached_util_bootstrap, temp_util_bootstrap])
+    # inspect the local packages cache to grab the desired artifact
+    cached_package_path = 'packages/cache/packages/' + launch_package_id + '.tar.xz'
```

Comment:
It'd be nice to reuse `PackageSet.get_package_path()` here, but its `__init__()` does a bunch of unrelated work :\